### ISSUE TITLE
Weights 

### ISSUE THREAD

**Issue by GuillaumeStaermanML** [GiulsLu/Sinkhorn-Barycenters#2]
Hello and congratulations for you work :),

I runned your code to find a barycenter between two gaussians following your function  'testBarycenter()' but i got weird results, did I do some wrong? I also tested several bary_init and Sinkhorn parameters but still not working.

<img width="387" alt="Capture d’écran 2021-03-24 à 09 58 00" src="https://user-images.githubusercontent.com/48350423/112282448-6c470780-8c87-11eb-806c-0442c833bf0b.png">

Here is the code i used:

################################################
################################################
import torch
import matplotlib.pyplot as plt
from Distribution.Distribution import Distribution
from Barycenter.Barycenter import Barycenter

torch.manual_seed(42)

n = 100
n_bary = 100
d = 2

mu_support = torch.randn((n,d))
mu_support2 = torch.randn((n,d)) + 10

mu0 = Distribution(mu_support)
mu1 = Distribution(mu_support2)

bary_init_support = torch.randn((n_bary,d)) 
bary_init = Distribution(bary_init_support)
bary_init.normalize()

bary = Barycenter([mu0,mu1],bary=bary_init, sinkhorn_tol = 0.1, sinkhorn_n_itr = 100)
barycenter = bary._computeSinkhorn()

plt.scatter(mu0.support[:,0], mu0.support[:,1], c='r', label='First distribution', s=1)
plt.scatter(mu1.support[:,0], mu1.support[:,1], c='b', label='Second distribution', s=1)
plt.scatter(barycenter[1][:n_bary], barycenter[1][n_bary:], c='k', label='Barycenter', s=1)
plt.legend()
################################################
################################################

Furthermore, Is it possible to fix weights of the barycenter (to uniform for example) and to optimize only on the support ?

**Comment by PierreColombo** [comment]
Hello,
I would be interested in reproducing the same experiment as well :)

**Comment by GiulsLu** [comment]
Hi both! in the code that you pasted, you are using 

barycenter = bary._computeSinkhorn().

this is just to check that the function _computeSinkhorn is running without errors (as explained in the tests folder, sorry if the file name "testBarycenter" was confusing). What you are plotting, is what the function computeSinkhorn() returns, i.e. the potentials!

To compute the barycenter, you have to follow the steps that you can find e.g. in "experiments/ellipses.py".

Check here for a mini-documentation https://github.com/GiulsLu/Sinkhorn-Barycenters/blob/master/documentation.md
